protocol/daemons/types/health_checkable.go

```diff
@@ -0,0 +1,113 @@
+package types
+
+import (
+	"fmt"
+	libtime "github.com/dydxprotocol/v4-chain/protocol/lib/time"
+	"sync"
+	"time"
+)
+
+const (
+	MaximumAcceptableUpdateDelay = 5 * time.Minute
+)
+
+// HealthCheckable is a common interface for services that can be health checked.
+type HealthCheckable interface {
+	// HealthCheck returns an error if a service is unhealthy. If the service is healthy, this method returns nil.
+	HealthCheck(provider libtime.TimeProvider) (err error)
+}
+
+// HealthCheckableImpl implements the HealthCheckable interface by tracking the timestamps of the last successful and
+// failed updates. If:
+// - no update has occurred
+// - the most recent update failed, or
+// - the daemon has not seen a successful update in at least 5 minutes,
+// the service is considered unhealthy.
+type HealthCheckableImpl struct {
+	sync.Mutex
+
+	// lastSuccessfulUpdate is the timestamp of the last successful update.
+	lastSuccessfulUpdate time.Time
+	// lastFailedUpdate is the timestamp of the last failed update. After the HealthCheckableImpl is initialized,
+	// this should never be a zero value.
+	lastFailedUpdate time.Time
```

Comment:
nit: I assume that `lastFailedUpdate` and `lastUpdateError` will be updated together. It could be useful to define a new struct to couple them

```
type updateTimeAndError struct {
  ...
}
```